src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerItem.java

```diff
@@ -0,0 +1,146 @@
+				new IllegalArgumentException("ItemStack " + itemStack + " is not an item handler"));
+	}
+	
+	private static IMusicPlaying getMusicPlaying(ItemStack itemStack)
+	{
+		return itemStack.getCapability(CapabilityMusicPlaying.MUSIC_PLAYING_CAPABILITY).orElseThrow(() ->
+				new IllegalArgumentException("ItemStack " + itemStack + " is not a music playing"));
+	}
+	
+	@Nullable
+	@Override
+	public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt)
+	{
+		return new CapabilityProviderMusicPlayer();
+	}
+	
+	public MusicPlayerItem(Builder builder, Properties properties)
+	{
+		super(builder, properties);
+	}
+	
+	@Override
+	public InteractionResultHolder<ItemStack> use(Level level, Player playerIn, InteractionHand handIn)
+	{
+		IMusicPlaying musicPlaying = getMusicPlaying(playerIn.getItemInHand(handIn));
+		IItemHandler itemStackHandlerMusicPlayer = getItemStackHandlerMusicPlayer(playerIn.getItemInHand(handIn));
+		
+		if(playerIn.isCrouching())
+		{
+			SoundManager soundManager = Minecraft.getInstance().getSoundManager();
+			MusicPlayerOnPlayerSoundInstance soundInstance;
+			Item item = itemStackHandlerMusicPlayer.getStackInSlot(0).getItem();
+			if(item instanceof CassetteItem cassette)
+			{
+				if(musicPlaying.getCurrentMusic() == null)
+				{
+					musicPlaying.setCurrentMusic(EnumCassetteType.getSoundEvent(cassette.cassetteID));
+					musicEffect(cassette.cassetteID, playerIn);
+					soundInstance = new MusicPlayerOnPlayerSoundInstance(playerIn, musicPlaying);
+					soundManager.play(soundInstance);
+				} else
+				{
+					musicEffect(EnumCassetteType.NONE, playerIn);
+					musicPlaying.setCurrentMusic(null);
+				}
+			}
+		}
+		//open the GUI if right-clicked
+		else if(!level.isClientSide)
+		{
+			NetworkHooks.openGui((ServerPlayer) playerIn, new SimpleMenuProvider((pContainerId, pInventory, pPlayer) ->
+					new MusicPlayerContainer(pContainerId, pInventory, itemStackHandlerMusicPlayer),
+					new TextComponent("Music Player")));
+		}
+		return super.use(level, playerIn, handIn);
+	}
+	
+	@Override
+	public boolean hurtEnemy(ItemStack stack, LivingEntity target, LivingEntity attacker)
+	{
+		Random r = attacker.level.getRandom();
+		double chanceToHit = r.nextFloat();
+		AttributeInstance attackerLuck = attacker.getAttribute(Attributes.LUCK);
+		AttributeInstance targetLuck = target.getAttribute(Attributes.LUCK);
+		double attackerLuckValue;
+		double targetLuckValue;
+		if(attackerLuck != null)
+			attackerLuckValue = attackerLuck.getValue();
+		else
+			attackerLuckValue = 0.0D;
+		
+		if(targetLuck != null)
+			targetLuckValue = targetLuck.getValue();
+		else
+			targetLuckValue = 0.0D;
+		
+		chanceToHit = chanceToHit - (attackerLuckValue / 10) + (targetLuckValue / 10);
+		Debug.debug("chanceToHit: " + chanceToHit + " applyingChance: " + applyingChance);
+		Debug.debug("attackerLuckValue: " + attackerLuckValue + " targetLuckValue: " + targetLuckValue);
```

Comment:
Don't use `Debug`. Use `System.out.println()` for debugging purposes, or a class-specific logger for long-term purposes.

Reply:
Ok